### ISSUE TITLE
remove_singletons in ColoredHyperGraph throws error.

### ISSUE THREAD

**Issue by gurug-dev** [pyt-team/TopoNetX#288]
What :

remove_singletons() method of CHG does not work, throws error during creating a new CHG using restrict_to_cells() method.

When: (example below)
```
        CHG = ColoredHyperGraph([[1, 2, 3], [2, 3, 4]], ranks=2)
        CHG.add_cell([9]) #add a singleton cell
        assert CHG.singletons() == [(frozenset({9}), 0)] #works
        CHG.remove_singletons() #Throws error!
```
KeyError: 'hyperedge (frozenset({1, 2, 3}), 0) is not in the complex'
The above error is thrown during runtime of restrict_to_cells.

How to reproduce:
- Create a CHG with singleton cells.
- Verify the singleton cells using CHG.singletons()
- Try to remove singletons using CHG.remove_singletons()

Places to fix:
restrict_to_cells method of ColoredHyperGraph

**Comment by USFCA-MSDS** [comment]
This is being addressed with PR #348

**Comment by mhajij** [comment]
Closed with PR #348
